Python/opcode_metadata.h

```diff
@@ -56,21 +56,21 @@ _PyOpcode_num_popped(int opcode, int oparg, bool jump) {
         case UNARY_INVERT:
             return 1;
         case BINARY_OP_MULTIPLY_INT:
-            return 2;
-        case BINARY_OP_MULTIPLY_FLOAT:
-            return 2;
+            return 2+2;
```

Comment:
I see a lot of these values changing. That seems unexpected.

Reply:
I think that's because this is implemented by counting the total stack inputs consumed by all the constituent uops, and similarly (but separately) the total outputs pushed to stack by all constituent uops. So when an instruction gets split into two uops, and the first one pushes two (internal) "outputs" to stack that are consumed by the second one, you end up adding two to both "inputs consumed" and "outputs pushed", but that cancels out so the overall stack effect of the opcode remains the same.

Reply:
Yeah, but I would expect this value to not exceed the total stack size at that point. Previously, the `_PyOpcode_num_popped` and `_PyOpcode_num_pushed` could be used individually, but now they are only correct if used together, back to back, to calculate a raw stack effect.

For instance, I'm pretty sure these functions can currently be used to track data-dependencies between instructions in a CFG without resorting to full abstract interpretation. This change breaks that.

Reply:
I think it should be possible to update the cases generator so it cancels out the "internal" stack effects of macro instructions... internally, and they don't show up in `_PyOpcode_num_popped` and `_PyOpcode_num_pushed`.

There are already macro instructions that currently break this, though (`LOAD_NAME` is one, since the PEP 695 implementation was merged), so it's not new with this PR.

Reply:
Thanks for calling this out. This is indeed unexpected. I'll look into how this is computed in the generator (it's been a while since I was in that code). If it's simple to fix I'll do it here, otherwise I'll plan a separate PR.

Reply:
I've commented on #105034.

Reply:
I have now fixed this. @brandtbucher Can you re-review?